EIPS/eip-9999.md

```diff
@@ -0,0 +1,92 @@
+---
+eip: 9999
+title: Block-Level Access Lists
+description: Enforced block-level access lists for parallalizing transaction validation
+author: Toni Wahrstätter (@nerolation), Dankrad Feist (@dankrad), Francesco D`Amato (@fradamt), Yoav Weiss (@yoavw), Alex Forshtat (@forshtat)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-block-level-access-lists/23337
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-03-31
+---
+
+## Abstract
+
+This EIP proposes adding block-level access lists. By including a complete list of all addresses and storage keys accessed during a block, along with their initial values, in the block header, we can improve execution efficiency and enable better parallelization.
+
+## Motivation
+
+Currently, transactions without an explicit transaction access list cannot be efficiently parallelized, as the execution engine cannot determine in advance which addresses and storage slots will be accessed. While transaction-level access lists exist, they are not enforced, making it difficult to optimize execution pipelines.
+
+By enforcing access lists at the block level, we shift the responsibility of declaring storage accesses to the block builder. This enables the execution engine to efficiently parallelize transaction execution, knowing in advance the exact scope of storage interactions for each transaction. 
+
+With access lists at the block level, transactions can be scheduled and executed in parallel, allowing for performance improvements.
```

Comment:
For this parallelization topic, has the re-introduction of intermediate state roots in the tx receipts also be studied/considered? If we consider a bulky block builder then this should be fine, any additional data reads should be fine for the bulky block builder (but should be prevented on local/home nodes)

I'm not entirely sure what impact this could have on parallelization, but for instance one could prove a single tx with merkle proofs against the previous tx receipt root and use this as some kind of witness or help speeding up things (this intermediate tx state root is kind of a shower thought :sweat_smile: )